Source/WebKit/UIProcess/Downloads/DownloadProxy.h

```diff
@@ -104,6 +104,10 @@ class DownloadProxy : public API::ObjectImpl<API::Object::Type::Download>, publi
     void setProgress(NSProgress *progress) { m_progress = progress; }
     NSProgress *progress() const { return m_progress.get(); }
 #endif
+#if PLATFORM(GTK) || PLATFORM(WPE)
+    void setDownloadIsCancelled(bool cancelled) { m_downloadIsCancelled = cancelled; }
+    bool downloadIsCancelled() const { return m_downloadIsCancelled; }
+#endif
```

Comment:
So the first thing I don't like is that these are platform-specific. It's better to compile this on all platforms even if it's only used by GTK/WPE, so I would remove all the PLATFORM() guards.

Next, we don't actually need the downloadIsCancelled() member function. That can be replaced by checking m_downloadIsCancelled directly.

Then I would suggest renaming `setDownloadIsCancelled` to simply `cancel`. You never need to un-cancel a download, so there's no need for the bool parameter. That can be removed.

This leaves one last problem: after my suggested changes, it's redundant with the existing cancel function! I think you can get rid of your new function and just write `m_downloadIsCancelled = true` at the top of DownloadProxy::cancel.